### PR TITLE
More content in the home page

### PR DESCRIPTION
**Pull request by matteofigus**
*Description of changes:*
Bring more content to the home page

*PR Checklist:*

- [x] All tests pass
- [x] Pre-commit checks pass

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,27 +6,106 @@
...
+[cost overview guide](docs/COST_OVERVIEW.md) to learn about its costs.
+
+> Warning: This project is currently in beta release
+>
+> - Actively developed by AWS
+> - Carefully review code and documentation if you intend to use the solution
+>   for production purposes
+> - Get in touch with the team via
+>   [Github Issues](https://github.com/awslabs/amazon-s3-find-and-forget/issues)
+>   for providing feedback and to ask any question. We appreciate your help
```
Comment (matteofigus) [inline README.md:38]: I tried to provide more comprehensive details about what beta means for us.

File: README.md [hunk 2]
```diff
@@ -6,27 +6,106 @@
...
+See the [user guide](docs/USER_GUIDE.md) to learn how to use the solution and
+the [API Specification](docs/api/README.md) to integrate the solution to your
+own applications.
+
+## Architecture
+
+The goal of the solution is to provide a secure, reliable, performant and cost
+effective tool for finding and removing individual records within objects stored
+in S3 buckets. In order to achieve these goals the solution has adopted the
+following design principles:
```
Comment (matteofigus) [inline README.md:56]: The design principles in the architecture page are actually a bit of a mixture of design principles and consequences of some of our design principles. I tried to re-organise them here. If this makes sense, then I can also copy and paste there.

File: README.md [hunk 3]
```diff
@@ -6,27 +6,106 @@
...
+> - Get in touch with the team via
+>   [Github Issues](https://github.com/awslabs/amazon-s3-find-and-forget/issues)
+>   for providing feedback and to ask any question. We appreciate your help
+
+## Usage
+
 The solution provides a web user interface, and a REST API to allow you to
 integrate it in your own applications.
 
+![HomePage Screenshot](docs/images/home-screenshot.png)
```
Comment (matteofigus) [inline README.md:45]: Not sure about this one. The blogpost may have more details about this, so for now I just put one screenshot. But I can also remove it if we think it doesn't add any value

File: README.md [hunk 4]
```diff
@@ -6,27 +6,106 @@
...
+   encryption is performed at all layers at rest and in transit, authentication
+   is provided out of the box, expiration of logs is easily configurable, and
+   private data is automatically obfuscated or irreversibly deleted as soon as
+   possible when persisting state.
+2. **Built to scale:** The system has been designed and battle tested for
+   performance and high availability with Data Lakes of TB/PB size.
+3. **Cost optimised:**
+   - **Perform work in batches:** Since the time complexity of removing a single
+     vs multiple records in a single object is practically equal and it is
+     common for data owners to have the legal requirement of removing data
```
Comment (HieronymusLex) [inline README.md:68]:
```suggestion
     common for data owners to have the requirement of removing data
```

File: README.md [hunk 5]
```diff
@@ -6,27 +6,106 @@
...
+     within a given _timeframe_, the solution is designed to allow the customer
```
Comment (HieronymusLex) [inline README.md:69]:
```suggestion
     within a given _timeframe_, the solution is designed to allow the solution operator
```

File: README.md [hunk 6]
```diff
@@ -6,27 +6,106 @@
...
+1. **Security first:** Every component implements least privilege access,
+   encryption is performed at all layers at rest and in transit, authentication
+   is provided out of the box, expiration of logs is easily configurable, and
+   private data is automatically obfuscated or irreversibly deleted as soon as
+   possible when persisting state.
+2. **Built to scale:** The system has been designed and battle tested for
+   performance and high availability with Data Lakes of TB/PB size.
+3. **Cost optimised:**
+   - **Perform work in batches:** Since the time complexity of removing a single
+     vs multiple records in a single object is practically equal and it is
```
Comment (HieronymusLex) [inline README.md:67]:
```suggestion
     vs multiple matches in a single S3 object is practically equal, and it is
```

File: README.md [hunk 7]
```diff
@@ -6,27 +6,106 @@
...
+     common for data owners to have the legal requirement of removing data
+     within a given _timeframe_, the solution is designed to allow the customer
+     to "queue" multiple matches to be removed in a single job.
+   - **Fail fast:** A deletion job takes place in two distinct phases: Find and
+     Forget. The Find phase queries the objects in your S3 data lakes to find
+     any objects which contain records where a specified column contains at
```
Comment (HieronymusLex) [inline README.md:73]:
```suggestion
     any objects which contain matches where a specified column contains at
```

File: README.md [hunk 8]
```diff
@@ -6,27 +6,106 @@
...
+     least one of record identifiers (known as **Match IDs**) in the deletion
+     queue. If any queries fail, the job will abandon as soon as possible and
+     the Forget phase will not take place. The Forget Phase takes the list of
+     objects returned from the Find phase, and deletes the relevant rows in only
```
Comment (HieronymusLex) [inline README.md:77]:
```suggestion
     objects returned from the Find phase, and deletes only the relevant rows in
```

File: README.md [hunk 9]
```diff
@@ -6,27 +6,106 @@
...
+   - **Find, then Forget:** The process of separating the two phases, as
+     opposite of processing the read/write on an object basis, is another cost
+     optimisation technique based on the fact columnar dense formats such as
+     Parquet allow to find matches with increased performance and cost
+     optimisation by reading only relevant columns instead of full content from
+     all columns. By processing only the relevant columns during the Find and
+     deletegating it to a serverless managed service such as Amazon Athena, we
+     can then process full rows only during the Forget phase, restricting the
+     most expensive operation to only the relevant objects.
+   - **Serverless:** Where possible, the solution only uses Serverless
```
Comment (HieronymusLex) [inline README.md:88]: we should probably state why Serverless optmizes costs here

File: README.md [hunk 10]
```diff
@@ -6,27 +6,106 @@
...
+     the Forget phase will not take place. The Forget Phase takes the list of
+     objects returned from the Find phase, and deletes the relevant rows in only
+     those objects.
+   - **Find, then Forget:** The process of separating the two phases, as
+     opposite of processing the read/write on an object basis, is another cost
+     optimisation technique based on the fact columnar dense formats such as
+     Parquet allow to find matches with increased performance and cost
+     optimisation by reading only relevant columns instead of full content from
+     all columns. By processing only the relevant columns during the Find and
+     deletegating it to a serverless managed service such as Amazon Athena, we
```
Comment (HieronymusLex) [inline README.md:85]:
```suggestion
     delegating the queries to a serverless managed service such as Amazon Athena, we
```

File: README.md [hunk 11]
```diff
@@ -6,27 +6,106 @@
...
+     Forget. The Find phase queries the objects in your S3 data lakes to find
+     any objects which contain records where a specified column contains at
+     least one of record identifiers (known as **Match IDs**) in the deletion
+     queue. If any queries fail, the job will abandon as soon as possible and
+     the Forget phase will not take place. The Forget Phase takes the list of
+     objects returned from the Find phase, and deletes the relevant rows in only
+     those objects.
+   - **Find, then Forget:** The process of separating the two phases, as
+     opposite of processing the read/write on an object basis, is another cost
+     optimisation technique based on the fact columnar dense formats such as
```
Comment (HieronymusLex) [inline README.md:81]:
```suggestion
     optimisation technique leveraging columnar dense formats such as
```

File: README.md [hunk 12]
```diff
@@ -6,27 +6,106 @@
...
+     Parquet allow to find matches with increased performance and cost
```
Comment (HieronymusLex) [inline README.md:82]:
```suggestion
     Parquet allow to efficiently find matches
```

File: README.md [hunk 13]
```diff
@@ -6,27 +6,106 @@
...
+     optimisation by reading only relevant columns instead of full content from
```
Comment (HieronymusLex) [inline README.md:83]:
```suggestion
    by reading only relevant columns instead of full content from
```

File: README.md [hunk 14]
```diff
@@ -6,27 +6,106 @@
...
+     to "queue" multiple matches to be removed in a single job.
+   - **Fail fast:** A deletion job takes place in two distinct phases: Find and
+     Forget. The Find phase queries the objects in your S3 data lakes to find
+     any objects which contain records where a specified column contains at
+     least one of record identifiers (known as **Match IDs**) in the deletion
+     queue. If any queries fail, the job will abandon as soon as possible and
+     the Forget phase will not take place. The Forget Phase takes the list of
+     objects returned from the Find phase, and deletes the relevant rows in only
+     those objects.
+   - **Find, then Forget:** The process of separating the two phases, as
```
Comment (HieronymusLex) [inline README.md:79]: I don't think Find, then Forget is the right subheader for this section. I'm wondering if we're just restating the benefit of data formats like Parquet here, which won't necessarily hold true for other data formats like CSV/JSON?

File: README.md (hunk 7 above)
Comment (HieronymusLex) [reply README.md:73]: or records might be fine, I think we're just jumping between terms a bit

File: README.md [hunk 15]
```diff
@@ -6,27 +6,106 @@
...
+     deletegating it to a serverless managed service such as Amazon Athena, we
+     can then process full rows only during the Forget phase, restricting the
+     most expensive operation to only the relevant objects.
+   - **Serverless:** Where possible, the solution only uses Serverless
+     components. All the components for Web UI, API and Deletion Jobs are
+     Serverless (for more information consult the
+     [Cost Overview guide](docs/COST_OVERVIEW.md)).
+4. **Easy to use and operate:** The solution is designed to interact directly to
+   Amazon S3 for high compatibility with existing toolsets. It provides a REST
+   API to allow you to integrate it in your own applications, and a easy to use
```
Comment (HieronymusLex) [inline README.md:94]: let's avoid the word easy

File: README.md [hunk 16]
```diff
@@ -6,27 +6,106 @@
...
+   UI interface. When performing deletion jobs, information is provided in
+   real-time to allow good observability. After the job completes, detailed
+   reports are available documenting all the actions performed to individual S3
+   Objects, and detailed error traces in case of failures to guide the user to
+   the troubleshooting process and easily identify remediation actions (for more
```
Comment (HieronymusLex) [inline README.md:99]:
```suggestion
   the troubleshooting process and identify remediation actions. For more
```

File: README.md [hunk 17]
```diff
@@ -6,27 +6,106 @@
...
+   information consult the [Troubleshooting guide](docs/TROUBLESHOOTING.md)).
```
Comment (HieronymusLex) [inline README.md:100]:
```suggestion
   information consult the [Troubleshooting guide](docs/TROUBLESHOOTING.md).
```

File: README.md [hunk 18]
```diff
@@ -6,27 +6,106 @@
...
+     optimisation by reading only relevant columns instead of full content from
+     all columns. By processing only the relevant columns during the Find and
+     deletegating it to a serverless managed service such as Amazon Athena, we
+     can then process full rows only during the Forget phase, restricting the
+     most expensive operation to only the relevant objects.
+   - **Serverless:** Where possible, the solution only uses Serverless
+     components. All the components for Web UI, API and Deletion Jobs are
+     Serverless (for more information consult the
+     [Cost Overview guide](docs/COST_OVERVIEW.md)).
+4. **Easy to use and operate:** The solution is designed to interact directly to
```
Comment (HieronymusLex) [inline README.md:92]: The term "easy" is a bit subjective here. This section is a bit of a blend of operational excellence (robust error handling, logging etc.) and extensible by via an API

File: README.md [hunk 19]
```diff
@@ -6,27 +6,106 @@
...
+
+The goal of the solution is to provide a secure, reliable, performant and cost
+effective tool for finding and removing individual records within objects stored
+in S3 buckets. In order to achieve these goals the solution has adopted the
+following design principles:
+
+1. **Security first:** Every component implements least privilege access,
+   encryption is performed at all layers at rest and in transit, authentication
+   is provided out of the box, expiration of logs is easily configurable, and
+   private data is automatically obfuscated or irreversibly deleted as soon as
```
[the remark this answers is not in the document]
Comment (matteofigus) [reply README.md:61]: what about MatchIDs

File: README.md (hunk 6 above)
Comment (matteofigus) [reply README.md:67]: Not sure, do you think that match could be confused with the idea of actually just removing that value and leaving the rest of the row there? Perhaps we can use `rows`?

File: README.md (hunk 7 above)
Comment (matteofigus) [reply README.md:73]: yeah, I think records is what we were using to call "rows" as opposite of match which is only the value for that column?

File: README.md (hunk 14 above)
Comment (matteofigus) [reply README.md:79]: Ok, perhaps we can rephrase the subheader as `Optimised for Parquet`?

Comment (HieronymusLex) [reply README.md:79]: yeah either that or "Optimised for Columnar Data Formats"

File: README.md (hunk 19 above)
[the remark this answers is not in the document]
Comment (HieronymusLex) [reply README.md:61]: yeah Match IDs is probably more consistent - have we introduced that term by this point in the document though?

File: README.md (hunk 7 above)
Comment (HieronymusLex) [reply README.md:73]: yeah that sounds right

File: README.md (hunk 9 above)
Comment (matteofigus) [reply README.md:88]: Done

File: README.md [hunk 20]
```diff
@@ -6,27 +6,106 @@
...
 Once configured, you can queue record identifiers that you want the
 corresponding data erased for. You can then run a deletion job to remove the
 data corresponding to the records specified from the objects in the data lake. A
 report log is provided of all the S3 objects modified.
 
+## Installation
+
+The solution is available as an AWS CloudFormation template and should take
+about 20 to 40 minutes to deploy. See the
+[deployment guide](docs/USER_GUIDE.md#deploying-the-solution) to launch the
```
[the remark this answers is not in the document]
Comment (matteofigus) [reply README.md]: Done

File: README.md (hunk 19 above)
[the remark this answers is not in the document]
Comment (matteofigus) [reply README.md]: We didn't, so I introduced the term there, hopefully it makes sense now

File: README.md (hunk 18 above)
Comment (matteofigus) [reply README.md]: Not sure here, Excellent to operate seems arrogant 😂  - what about "Simple" 😂

Comment (HieronymusLex) [reply README.md]: https://aws.amazon.com/architecture/well-architected/ - see number 2

The problem I have with terms like easy and simple is that they are subjective based on the experience of the operator

Comment (matteofigus) [reply README.md]: Ok, I changed to `Ready to use and operate`? I think that is quite equivalent to what I'm trying to express with the paragraph?

File: README.md (hunk 3 above)
Comment (ctd) [reply README.md:47]: That screenshot on its own doesn't feel very valuable to me.

I have no problem with adding it now, but I think we should improve it ... maybe experiment with creating an animated GIF that shows the typical workflow?

Comment (matteofigus) [reply README.md]: I think you're right. I removed it for now, let's try making a gif in a separate PR

**Pull request merged**